src/options.js

```diff
@@ -88,13 +88,19 @@ export function arrayify(data, type) {
 // An optimization of type.from(values, f): if the given values are already an
 // instanceof the desired array type, the faster values.map method is used.
 export function map(values, f, type = Array) {
-  return values instanceof type ? values.map(f) : type.from(values, f);
+  if (values instanceof type) return values.map(f);
```

Comment:
Do we know that `values` is an array here? It could be that we are using `map` for non-arrays like iterables and {length} objects too.